src/qcodes/dataset/dond/do_0d.py

```diff
@@ -5,88 +5,43 @@
 
-from qcodes import config
-from qcodes.parameters import ParameterBase
+from .do_nd import DondKWargs, dond
 
-from ..descriptions.detect_shapes import detect_shape_of_measurement
-from ..measurements import Measurement
-from ..threading import process_params_meas
-from .do_nd_utils import _handle_plotting, _register_parameters, _set_write_period
+if TYPE_CHECKING:
+    from typing_extensions import Unpack
+
+    from .do_nd_utils import (
+        AxesTupleListWithDataSet,
+        ParamMeasT,
+    )
 
 LOG = logging.getLogger(__name__)
 TRACER = trace.get_tracer(__name__)
 
-if TYPE_CHECKING:
-    from ..descriptions.versioning.rundescribertypes import Shapes
-    from ..experiment_container import Experiment
-    from .do_nd_utils import AxesTupleListWithDataSet, ParamMeasT
-
 
 @TRACER.start_as_current_span("qcodes.dataset.do0d")
 def do0d(
-    *param_meas: ParamMeasT,
-    write_period: float | None = None,
-    measurement_name: str = "",
-    exp: Experiment | None = None,
-    do_plot: bool | None = None,
-    use_threads: bool | None = None,
-    log_info: str | None = None,
+    *param_meas: ParamMeasT, **kwargs: Unpack[DondKWargs]
 ) -> AxesTupleListWithDataSet:
     """
     Perform a measurement of a single parameter. This is probably most
-    useful for an ArrayParameter that already returns an array of data points
+    useful for a ParameterWithSetpoints that already returns an array of data points.
 
     Args:
         *param_meas: Parameter(s) to measure at each step or functions that
           will be called at each step. The function should take no arguments.
           The parameters and functions are called in the order they are
           supplied.
-        write_period: The time after which the data is actually written to the
-            database.
-        measurement_name: Name of the measurement. This will be passed down to
-            the dataset produced by the measurement. If not given, a default
-            value of 'results' is used for the dataset.
-        exp: The experiment to use for this measurement.
-        do_plot: should png and pdf versions of the images be saved after the
-            run. If None the setting will be read from ``qcodesrc.json``
-        use_threads: If True measurements from each instrument will be done on
-            separate threads. If you are measuring from several instruments
-            this may give a significant speedup.
-        log_info: Message that is logged during the measurement. If None a default
-            message is used.
+        **kwargs: kwargs are the same as for dond and forwarded directly to dond.
 
     Returns:
         The QCoDeS dataset.
 
     """
-    if do_plot is None:
-        do_plot = cast("bool", config.dataset.dond_plot)
-    meas = Measurement(name=measurement_name, exp=exp)
-    if log_info is not None:
-        meas._extra_log_info = log_info
-    else:
-        meas._extra_log_info = "Using 'qcodes.dataset.do0d'"
```

Comment:
shall we keep track of whether do0d was called epxlicitly by passing do0d's default log_info to the underlying dond? If someone in the commiunity still relies on this, it could be a 'breaking-in-sense' change. Or do we only rely on the better `@TRACER.start_as_current_span("qcodes.dataset.do0d")`?

Reply:
Agreed. Done both for do1d and do0d